victron_ble/devices/solar_charger.py

```diff
@@ -0,0 +1,90 @@
+from enum import Enum
+from typing import Any, Dict
+
+from construct import Int16sl, Int16ul, Struct
+
+from victron_ble.devices.base import Device, DeviceData
+
+
+class ChargeState(Enum):
+    OFF = 0
+    BULK = 3
+    ABSORPTION = 4
+    FLOAT = 5
+
+
+class SolarChargerData(DeviceData):
+    def __init__(self, data: Dict[str, Any]) -> None:
+        self._data = data
+
+    def get_charge_state(self) -> ChargeState:
+        """
+        Return an enum indicating the current meter type
+        """
+        return self._data["charge_state"]
+
+    def get_voltage(self) -> float:
+        """
+        Return the voltage in volts
+        """
+        return self._data["voltage"]
+
+    def get_current(self) -> float:
+        """
+        Return the current in amps
+        """
+        return self._data["current"]
+
+    def get_yield_today(self) -> float:
+        """
+        Return the yield_today in Wh
+        """
+        return self._data["yield_today"]
+
+    def get_power(self) -> float:
+        """
+        Return the current solar power in W
+        """
+        return self._data["power"]
+
+    def get_load(self) -> float:
+        """
+        Return the load in amps
+        """
+        return self._data["load"]
+
+
+class SolarCharger(Device):
+
+    PACKET = Struct(
+        # state:    0 - Off
+        #           3 - Bulk
+        #           4 - Absorption
+        #           5 - Float
+        "charge_state" / Int16sl,
+        # Voltage reading in 0.1v increments
```

Comment:
0.01v increments

Reply:
I was first thinking the same, but it's not. It seems to be only 0.1V increments. Unfortunately it's already dark over here, so I can not provide any sample data.

Reply:
Just thinking if my comments are wrong. This is how I calc the voltage:

 "voltage": pkt.voltage / 100

Reply:
It's sunny in Trinidad:
```
Container: 
    state = 3
    battery_voltage = 2577
    current = 182
    yield_today = 76
    power = 482
    unknown = 255
    unknown2 = 255
    unknown3 = 69
    unknown4 = 107
    unknown5 = 146
    unknown6 = 22
```

Reply:
I fixed the my comments...

Reply:
I think my original comment in the battery meter was wrong, this is indeed 0.01v increments

Reply:
Filed #9 to fix the comments